packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -91,139 +117,535 @@ const propTypes = {
+    // When 'Predictive text' is enabled on Android the maxLength attribute on the input is ignored so we have to
+    // check the length of inputValue to make sure that it is less then 2.
+    if (inputValue === date.day || inputValue.length > 2 || Number(inputValue) > 31 || inputValue === '00') {
+      return;
+    }
+
+    // If the change made was not a deletion of a digit, then prepend '0'
+    // when the input value is a single digit value between 4 and 9
+    if (inputValue.length >= date.day.length) {
+      const digitsToPrependZero = ['4', '5', '6', '7', '8', '9'];
 
-    return (
-      <div className={cx(theme.className)}>
-        <div className={cx('custom-input')}>
+      if (digitsToPrependZero.indexOf(inputValue) > -1) {
+        inputValue = `0${inputValue}`;
+      }
+    }
+
+    handleDateChange(event, inputValue, DateUtil.inputType.DAY);
+  };
+
+  const handleMonthChange = (event) => {
+    let inputValue = event.target.value;
+    if (!DateUtil.validDateInput(inputValue)) {
+      return;
+    }
+
+    // Ignore the entry if the value did not change or it is invalid.
+    // When 'Predictive text' is enabled on Android the maxLength attribute on the input is ignored so we have to
+    // check the length of inputValue to make sure that it is less then 2.
+    if (inputValue === date.month || inputValue.length > 2 || Number(inputValue) > 12 || inputValue === '00') {
+      return;
+    }
+
+    // If the change made was not a deletion of a digit, then prepend '0'
+    // when the input value is a single digit value between 2 and 9
+    if (inputValue.length >= date.month.length) {
+      const digitsToPrependZero = ['2', '3', '4', '5', '6', '7', '8', '9'];
+
+      if (digitsToPrependZero.indexOf(inputValue) > -1) {
+        inputValue = `0${inputValue}`;
+      }
+    }
+
+    handleDateChange(event, inputValue, DateUtil.inputType.MONTH);
+  };
+
+  const handleYearChange = (event) => {
+    const inputValue = event.target.value;
+    if (!DateUtil.validDateInput(inputValue)) {
+      return;
+    }
+
+    // Ignore the entry if the value did not change or it is invalid.
+    // When 'Predictive text' is enabled on Android the maxLength attribute on the input is ignored so we have to
+    // check the length of inputValue to make sure that it is less then 4.
+    if (inputValue === date.year || inputValue.length > 4) {
+      return;
+    }
+
+    // Ignore the entry if the year value is not between 1900 and 2100
+    if (inputValue.length === 4 && (Number(inputValue) < 1900 || Number(inputValue) > 2100)) {
+      return;
+    }
+
+    handleDateChange(event, inputValue, DateUtil.inputType.YEAR);
+  };
+
+  const setInputFocus = (event, inputRef, selectionStart, selectionEnd) => {
+    inputRef.focus();
+    inputRef.setSelectionRange(selectionStart, selectionEnd);
+    event.preventDefault();
+  };
+
+  const handleDayInputKeydown = (event) => {
+    if (inputAttributes?.readOnly) {
```

Comment:
This chaining operator needs to be changed (wherever used) as it is throwing error. Reference [comment](https://github.com/cerner/terra-toolkit-boneyard/pull/75#discussion_r546002269)

Reply:
I don't think we need to change it here as babel automatically expands it during compilation.